### PR TITLE
[b/385205338] Add Bigtable Readme

### PR DESCRIPTION
**Pull request by InigoSJ**
Add readme for the Bigtable Sink connector

fixes b/385205338

### CODE REVIEW

File: connectors/bigtable/README.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+*   `key` (String)
+*   `family1` (contains fields like `name` (String), `age` (Integer))
+*   `family2` (contains fields like `city` (String), `zipCode` (Integer)) 
+
+Column Family `family1` would have columns `name, age` and Column Family `family2` would have columns `city, zipCode`.
+
+### Supported Types
+
+Both `RECORD` and `ROW` are not supported unless it's to define a Column Family in `Nested Rows Mode`. Double nested are not supported. For those and other unsupported types, you can use `BYTES`.
```
Comment (clmccart) [inline connectors/bigtable/README.md:92]: nit: "Double nested rows"

File: connectors/bigtable/README.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+| `project` | (Required) The Google Cloud project ID. |
+| `instance` | (Required) The Bigtable instance ID. |
+| `table` | (Required) The Bigtable table ID. |
+| `column-family` | The column family to write to (incompatible with `use-nested-rows-mode`). |
+| `use-nested-rows-mode` | Whether to use nested rows as column families (incompatible with `column-family`). |
+| `sink.parallelism` | The parallelism of the sink. |
+
+## Exactly Once
+
+Due to Bigtable's nature, the Sink offers Exactly Once out of the box. In order to get Exactly Once, every element needs to have a timestamp which will be used in the `RowMutationEntry`.
```
Comment (clmccart) [inline connectors/bigtable/README.md:168]: have we confirmed this? ie, verified correctness / fault tolerance if TMs go down?

File: connectors/bigtable/README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+| Incoming Mutation (in order)                                  | Bigtable State (after mutation, at ts NOW) |
+| :------------------------------------------------------------ | :---------------------------------------- |
+| RowKey, CF, C, timestamp1, my-value                           | my-value                                   |
+| RowKey, CF, C, timestamp2, my-value-2                         | my-value2                                  |
+| RowKey, CF, C, timestamp1, my-value                           | my-value2                                  |
+| RowKey, CF, C, timestamp3, my-value                           | my-value                                   |
+
+This means that, as long as the timestamp doesn't change with retries (i.e., it is idempotent with retries), Exactly Once works the same as At Least Once.
```
Comment (clmccart) [inline connectors/bigtable/README.md:179]: we should include an example of how to ensure that the timestamp is applied to the element for exactly-once

File: connectors/bigtable/README.md (hunk 2 above)
Comment (InigoSJ) [reply connectors/bigtable/README.md:168]: Talked offline.

This was verified by checking the state of the reads by inserting data to Bigtable through the client libraries but not through the Sink itself.

Sink doesn't add any special recovery but relies on BT's idempotency + Flink's EO

Talking with the Bigtable team, this is right.

Anyhow, testing will be done

File: connectors/bigtable/README.md (hunk 3 above)
Comment (InigoSJ) [reply connectors/bigtable/README.md:179]: Talked offline. WE are doing this in a separate PR

File: connectors/bigtable/README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+```
+
+In this mode, all fields except `RowKeyField` must be Rows. The Row name represent the Column Family. For example:
+
+*   `key` (String)
+*   `family1` (contains fields like `name` (String), `age` (Integer))
+*   `family2` (contains fields like `city` (String), `zipCode` (Integer)) 
+
+Column Family `family1` would have columns `name, age` and Column Family `family2` would have columns `city, zipCode`.
+
```
[the remark this answers is not in the document]
Comment (clmccart) [reply connectors/bigtable/README.md:91]: ah okay so users can expect an error in this case

**Review by clmccart — DISMISSED** [review]

**Review by clmccart — APPROVED** [review]

**Pull request merged**